ast/ddl.go

```diff
@@ -1728,6 +1728,7 @@ const (
 	AlterTableCoalescePartitions
 	AlterTableDropPartition
 	AlterTableTruncatePartition
+	AlterTableTruncateAllPartition
```

Comment:
See [MySQL Doc](https://dev.mysql.com/doc/refman/8.0/en/alter-table.html),   the `ALL` option is used wildly in `Partitions Operations` in `ALTER TABLE` syntax. I think  it's better to reuse `AlterTableTruncatePartition` better than add another operation type `AlterTableTruncateAllPartition` . There may be many duplicates otherwise.  You may add an `OperateOnAllPartition` flag or so in struct `AlterTableSpec` to indicate this, I think

Reply:
But as [this](https://pingcap.com/blog-cn/30mins-become-contributor-of-tidb-20190808/) suggests, we should follow the principle of **"try not to modify the structure of AST"**.

Reply:
I think the `principle` means we should not add extra `AST` node type or do something break the correctness of current code, not prevent us from doing any modification on current code. 
What do you think? @leoppro @kennytm

Reply:
I agree with @spongedu

Reply:
OK, I'll fix it.